common/test/utils.py

```diff
@@ -117,19 +116,22 @@ def skip_signal(signal, **kwargs):
         signal.connect(**kwargs)
 
 
-class MockS3Mixin(object):
+class MockS3BotoMixin(object):
     """
-    TestCase mixin that stubs S3 using the moto library. Note that this will
-    activate httpretty, which will monkey patch socket.
+    TestCase mixin that mocks the S3BotoStorage save method and s3 connection.
     """
     def setUp(self):
-        super(MockS3Mixin, self).setUp()
-        self._mock_s3 = moto.mock_s3_deprecated()
-        self._mock_s3.start()
+        super(MockS3BotoMixin, self).setUp()
+
+        with patch('boto.connect_s3') as self.mocked_connection:
+            self.mocked_connection.return_value = Mock()
```

Comment:
Doesn't this patch stay active only for the duration of this `with` statement?  I thought you'd need another patcher object like the one below, started here and stopped in `tearDown()`.